### PR TITLE
Mccalluc/metadata

### PR DESCRIPTION
**Pull request by mccalluc**
I believe this should support numeric metadata files. This felt kind-of spaghetti-ish, but I'm not sure that there's anything obvious in this commit that should have been done differently. I think some time on cypress testing would useful, as the number of options in the UI expands.

### CODE REVIEW

**Comment by mccalluc** [comment]
Fix #137; towards #140.

File: context/app/utils/callbacks.py [hunk 1]
```diff
@@ -52,22 +53,43 @@ def traces_all_selected(x_axis, y_axis, everyone, selected,
...
         {
             'name': 'gene axis',
             'dataframe': highlight,
             'marker': _big_light_dot
-        },
-        {
+        }
+    ]
+    if color_by is not None:
+        trace_defs.append({
```
Comment (scottx611x) [inline context/app/utils/callbacks.py]: Maybe .extend() is more appropriate here?

File: context/app/vis/condition_callbacks.py [hunk 2]
```diff
@@ -44,10 +48,14 @@ def _update_scatter_pca(
             highlight = everyone.loc[[gene_x_axis, gene_y_axis]]
             selected_highlight = everyone.loc[
                 list(set(selected.index) & set(highlight.index))
-            ]  # pandas.merge loses the index, sadly, so pick a hack.
-            data = traces_all_selected(x_axis, y_axis, everyone, selected,
-                                       highlight=highlight,
-                                       selected_highlight=selected_highlight)
+            ]  # pandas.merge loses the index, sadly, so this is a hack.
+            data = traces_all_selected(
```
Comment (scottx611x) [inline context/app/vis/condition_callbacks.py]: This function is beginning to have a lot of arguments, I'm sure theres ways around it, but then again the intentions are clear as it is. Probably not a big deal

File: context/app/utils/callbacks.py (hunk 1 above)
Comment (mccalluc) [reply context/app/utils/callbacks.py]: good call

File: context/app/vis/condition_callbacks.py (hunk 2 above)
Comment (mccalluc) [reply context/app/vis/condition_callbacks.py]: It's something I'll keep an eye on. Possibilities:
- Wrapper objects that encapsulate several fields each?
- Split up the method?

**Pull request merged**